server/TypeDBService.java

```diff
@@ -380,35 +386,33 @@ public void sessionOpen(SessionProto.Session.Open.Req request,
             int duration = (int) Duration.between(start, Instant.now()).toMillis();
             responder.onNext(openRes(sessionSvc.UUID(), duration));
             responder.onCompleted();
-            Diagnostics.get().requestSuccess(SESSION);
+            Diagnostics.get().requestSuccess(request.getDatabase(), SESSION_OPEN);
         } catch (RuntimeException e) {
             LOG.error(e.getMessage(), e);
-            Diagnostics.get().submitError(e);
+            Diagnostics.get().requestFail(request.getDatabase(), SESSION_OPEN);
+            Diagnostics.get().submitError(request.getDatabase(), e);
             responder.onError(exception(e));
-        } finally {
-            Diagnostics.get().setCurrentCount(SESSIONS, sessionServices.size());
         }
     }
 
     @Override
     public void sessionClose(SessionProto.Session.Close.Req request,
                              StreamObserver<SessionProto.Session.Close.Res> responder) {
+        String databaseName = null;
         try {
-            Diagnostics.get().requestAttempt(SESSION);
             UUID sessionID = byteStringAsUUID(request.getSessionId());
             SessionService sessionSvc = sessionServices.get(sessionID);
             if (sessionSvc == null) throw TypeDBException.of(SESSION_NOT_FOUND, sessionID);
+            databaseName = sessionSvc.session().database().name();
```

Comment:
Not as careful here as in TransactionService, eh?

Reply:
Well, it's inside `try` block and it would be a valid throw if the request didn't contain any of `session`, `database`, or its `name`. That's what I thought.

Reply:
Hmm but now if `sessionSvc != null`, but `database` for some reason is, `sessionSvc.close()` won't be called. I don't know if that's even possible, but it is changing the behaviour in that case.

Reply:
You're right, I didn't pay enough attention to the business logic here, my bad. And having a `throw` because of this line after everything would be dumb as well. Will be careful here as well then...

Reply:
Let's actually add a `databaseName()` getter to the session service so that we can at least have one fewer null check in these places.

Reply:
Good idea, done.